terraform/modules/storage/main.tf

```diff
@@ -8,6 +8,21 @@ resource "google_storage_bucket" "phi_storage_bucket" {
   storage_class = "MULTI_REGIONAL"
 }
 
+locals {
+  pipeline_modes = ["source-data", "failed_fhir_conversion", "failed_fhir_upload"]
+  message_types  = ["elr", "vxu", "ecr"]
+
+  phi_directories = { for directory in setproduct(local.pipeline_modes, local.message_types) :
+  "${directory[0]}/${directory[1]}/" => directory }
+}
+
+resource "google_storage_bucket_object" "phi_folders" {
+  for_each = local.phi_directories
+  name     = each.key
+  bucket   = google_storage_bucket.phi_storage_bucket.name
+  content  = "empty directory"
```

Comment:
It seems to be a little bit of a hack. Normally it would create a file with the contents "empty directory", but because the file names ends with "/" GCP doesn't actually make a proper file. Terraform requires that there is a values for contents and tried just using an empty string, but that wasn't sufficient.